companyweb_payment_info/wizards/payment_info_wizard.py

```diff
@@ -0,0 +1,393 @@
+            period_to_send.month,
+            period_to_send.year,
+        )
+        if response_previous_period["PreviousPeriodExists"]:
+
+            summary = _(
+                "<h2>Companyweb upload</h2>"
+                "You are about to submit <strong>%s</strong> open invoices <br/>"
+                "to Companyweb for the company <strong>%s</strong>.<br/>"
+                "The previous period that was sent is <strong>%s</strong>.<br/>"
+                "Your odoo login : <strong>%s</strong> will receive a full "
+                "resume at the end of this transaction <br/>"
+                "Make sure you have closed your bank statements for period <strong>%s</strong>."
+                % (
+                    len(invoice_to_send),
+                    html.escape(self.env.user.company_id.name),
+                    last_period_sent,
+                    self.env.user.partner_id.email,
+                    print_period_to_send,
+                )
+            )
+        else:
+            summary = _(
+                "<h2>Companyweb upload</h2>"
+                "You are about to submit <strong>%s</strong> open invoices <br/>"
+                "to Companyweb for the company <strong>%s</strong>.<br/>"
+                "<strong>It's the first time you use this feature "
+                "for this company</strong><br/>"
+                "Your odoo login : <strong>%s</strong> will received a full resume "
+                "at the end of this transaction <br/>"
+                "Make sure you have closed your bank statements for "
+                "period <strong>%s</strong>."
+                % (
+                    len(invoice_to_send),
+                    html.escape(self.env.user.company_id.name),
+                    self.env.user.login,
+                    print_period_to_send,
+                )
+            )
+        return summary
+
+    def _create_invoices_to_send(self, client):
+        open_invoices = self._get_open_invoices()
+        if not open_invoices:
+            return []
+
+        invoice_request_type = client.get_type("ns0:InvoiceRequest")
+        invoice_to_send = []
+
+        for invoice in open_invoices:
+            if not invoice.partner_id.vat:
+                continue
+            if not VAT_RE.match(invoice.partner_id.vat):
+                continue
+
+            to_send = invoice_request_type(
+                InvoiceNumber=invoice.id,
+                InvoiceDate=invoice.invoice_date.strftime("%Y%m%d"),
+                ExpirationDate=invoice.invoice_date_due.strftime("%Y%m%d"),
+                OpenAmount=invoice.amount_total_signed,
+                VatNumber=invoice.partner_id.vat,
+            )
+            invoice_to_send.append(to_send)
+        return invoice_to_send
+
+    def _get_open_invoices(self):
+        return self.env["account.move"].search(
+            [
+                ("payment_state", "=", "not_paid"),
+                ("state", "=", "posted"),
```

Comment:
Don't we need to filter on customer invoices only ?
What about customer refunds ?

Reply:
Filter only on out_invoice done